### ISSUE TITLE
Document IArgumentCompleter and ArgumentCompleterAttribute for binary module development

### ISSUE THREAD

**Issue by rjmholt** [MicrosoftDocs/PowerShell-Docs#6577]
Argument completers can be defined in .NET DLLs by implementing the `IArgumentCompleter` interface and using the `ArgumentCompleterAttribute` on parameters and cmdlets.

There are examples of this in the wild, but there doesn't seem to be any documentation of it beyond API documentation.

We should document how these types can be used not just from PowerShell but from binary modules and give examples. The PowerShell codebase should probably also improve the doc comments on the APIs.

**Comment by sdwheeler** [comment]
@rjmholt I agree, we should have some docs that provide examples of using this attribute. 

Note that the API docs are autogenerated from /// comments in the source code. So any of those doc changes need to be made there.
